src/Sentry/Scope.cs

```diff
@@ -112,8 +112,33 @@ public User User
         /// <inheritdoc />
         public string? Environment { get; set; }
 
+        // TransactionName and Transaction need to coexist because
+        // SentryEvent still tracks the transaction as a string.
+        // If a started Transaction exists and the user sets TransactionName,
+        // it should update the name of that transaction.
+        // Otherwise, it should just keep track of the name separately, so that
+        // it can be used in an event, in case a transaction doesn't exist.
+        // More context in this discussion:
+        // https://github.com/getsentry/develop/issues/246#issuecomment-760997974
+
+        private string? _fallbackTransactionName;
+
         /// <inheritdoc />
-        public string? TransactionName { get; set; }
+        public string? TransactionName
```

Comment:
I know you're waiting for a reply from @rhcarvalho on that thread but if we're going ahead with this, there are a few test cases here right?